src/browser/BLECentralPlugin.js

```diff
@@ -1,63 +1,194 @@
+        navigator.bluetooth.requestDevice(options).then(device => {
+            if (this.scanning) {                
+                success(device);
+            }
+        }).catch(failure);
     },
     connect: function(device_id, connectSuccess, connectFailure) {
-        notSupported();
-        if (connectFailure) connectFailure();
-    },
-    disconnect: function(device_id, connectSuccess, connectFailure) {
-        notSupported();
-        if (connectFailure) connectFailure();
+        console.log('connect', device_id);
+        if (this.devices.has(device_id)) {
+            connectSuccess(this.devices.get(device_id).device);
+        } else {
+            navigator.bluetooth.requestDevice({filters: [{
+                services: [getDeviceId(device_id)]
+            }]}).then(device => {
+                return device.gatt.connect().then(server => {
+                    this.devices.set(device_id, {
+                        device: device,
+                        server: server
+                    })
+                    connectSuccess(device);
+                });
+            }).catch(connectFailure);
+        }
+    },
+    disconnect: function(device_id, disconnectSuccess, disconnectFailure) {
+        if (this.devices.has(device_id)) {
+            var device = this.devices.get(device_id).device;
+            this.devices.delete(device_id);
+            if (device.gatt.connected) {
+                device.gatt.disconnect();
+                disconnectSuccess(device);
+            } else {
+                disconnectFailure("Device not connected");
+            }
+        } else if (disconnectFailure) {
+            disconnectFailure();
+        }
     },
     read: function(device_id, service_uuid, characteristic_uuid, success, failure) {
-        notSupported();
-        if (failure) failure();
+       if (this.devices.has(device_id)) {
+            this.devices.get(device_id).server.getPrimaryService(service_uuid).then(service => {
+                return service.getCharacteristic(characteristic_uuid);
+            }).then(characteristic => {
+                return characteristic.readValue();
+            }).then(result => {
+                success(result);
+            }).catch(error => {
+                if (failure) failure(error);
+            });
+        } else if (failure) { 
+          failure();
+        }
     },
     readRSSI: function(device_id, success, failure) {
         notSupported();
         if (failure) failure();
     },
     write: function(device_id, service_uuid, characteristic_uuid, data, success, failure) {
-        notSupported();
-        if (failure) failure();
+        if (this.devices.has(device_id)) {
+            this.devices.get(device_id).server.getPrimaryService(service_uuid).then(service => {
+                return service.getCharacteristic(characteristic_uuid);
+            }).then(characteristic => {
+                return characteristic.writeValue(data);
```

Comment:
I'd suggest this uses `writeValueWithResponse` to align with the other platforms (https://developer.mozilla.org/en-US/docs/Web/API/BluetoothRemoteGATTCharacteristic/writeValueWithResponse)